### PR TITLE
SALTO-5692, SALTO-6358: Okta: Upgrade `ApplicationGroupAssignment` deploy definitions to new infra

### PR DESCRIPTION
**Pull request by Nurdok**
This PR adds HTTP-recording tests to `ApplicationGroupAssignment` _and_ upgrades this type to the new infra. This also fixes a bug where `modify` always fails with HTTP 400 status code due to `appId` and `groupId` not being omitted.

---

_Additional context for reviewer_:
None

---
_Release Notes_: 
_Okta adapter_:
- Fixed a bug where modifying `ApplicationGroupAssignment` failed with a 400 HTTP status code.

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/okta-adapter/src/definitions/requests/clients.ts [hunk 1]
```diff
@@ -87,6 +87,11 @@ export const createClientDefinitions = (
               omitBody: true,
             },
           },
+          '/api/v1/apps/{appId}/groups/{id}': {
```
Comment (shir-reifenberg) [inline packages/okta-adapter/src/definitions/requests/clients.ts:90]: Should we add it under `default` ?

File: packages/okta-adapter/test/adapter.test.ts [hunk 2]
```diff
@@ -1112,23 +1122,123 @@ describe('adapter', () => {
...
+      })
+
+      it('should successfully add an app group assignment', async () => {
+        loadMockReplies('app_group_assignment_add.json')
+        const appGroupAssignment = new InstanceElement(
+          'appGroupAssignment',
+          appGroupAssignmentType,
+          {
+            id: group.value.id,
+            priority: 2,
```
Comment (shir-reifenberg) [inline packages/okta-adapter/test/adapter.test.ts:1158]: nit: let's add the `groupId` and `appId` fields to value to verify this test will break if somehow the code that currently omits those will change?

**Review by shir-reifenberg — APPROVED** [review]
Looks great with 1 question 😃

File: packages/okta-adapter/src/definitions/requests/clients.ts (hunk 1 above)
Comment (Nurdok) [reply packages/okta-adapter/src/definitions/requests/clients.ts:90]: I'll add as default in a separate PR.

**Pull request merged**